capstone/src/test/java/com/step902020/capstone/UserDatastoreTest.java

```diff
@@ -89,4 +96,122 @@ public void testGetOrganization() throws URISyntaxException {
 
     Assert.assertEquals("Wrong user returned", expectedEmail, result[0].getEmail());
   }
+
+  @Test
+  public void testAddEvent() throws URISyntaxException {
+
+    String url = "http://localhost:" + port + "/add-saved-event";
+    String expectedEmail = expectedIndividual.getEmail();  
+ 
+    restTemplate = new TestRestTemplate();
+    HttpHeaders headers = new HttpHeaders();
+    headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+
+    MultiValueMap<String, String> map= new LinkedMultiValueMap<String, String>();
+    map.add("email", expectedEmail);
+    map.add("event-name", "test");
+
+    HttpEntity<MultiValueMap<String, String>> request = new HttpEntity<MultiValueMap<String, String>>(map, headers);
+
+    ResponseEntity<String> response = restTemplate.postForEntity( url, request , String.class);
+
+    // getting the actual result
+    final String baseUrl = "http://localhost:"+ port + "/get-individual?email=" + expectedEmail;
+    URI uri = new URI(baseUrl);
+
+    Set<String> expectedSet = new HashSet<>();
+    expectedSet.add("test");
+ 
+    Individual[] result = restTemplate.getForObject(uri, Individual[].class);
+
+    Assert.assertEquals("Insert event error", expectedSet, result[0].getSavedEvents());
+  }
+
+  @Test
+  public void testDeleteEvent() throws URISyntaxException {
```

Comment:
It might not be immediately obvious, but there is an opportunity in these tests to make them less repetitive with helper functions. Notice the lines that are identical or almost identical across the test methods.

Reply:
I actually have some issues with these tests. I changed all the mappings to take in a RequestHeader param from the iap email header. But that causes these tests to fail locally. I have commented them out for now and I will create an issue to address these repetitive lines.